### PR TITLE
Add support for multiprocessing in PESQ

### LINKED ISSUE #1223: Slower performance for Torchmetrics PESQ
## 🐛 Bug

Multiprocessing capability is missing for Torchmetrics PESQ while the underlying library ludlows/python-pesq have that option.
As a result, the torchmetrics version of PESQ performs slower than ludlows/python-pesq.

### To Reproduce

Run PESQ for a batch of 100 audios with a minimum duration of 10 seconds for each audio.

#### Code sample

Python notebook code attachment here: 
[analysis.zip](https://github.com/Lightning-AI/metrics/files/9563279/analysis.zip)

### Expected behavior

Both the Torchmetrics PESQ and ludlows/python-pesq should take almost the same time to compute the PESQ score.

### Environment

- Python - 3.10.4 (installed using conda)
- OS: Ubuntu 20.04.4 LTS

Below libraries installed using pip
- TorchMetrics - 0.9.3
- PyTorch - 1.12.1
- Torchaudio - 0.12.1
- PESQ - 0.0.4

### Additional context

References
ludlows/python-pesq: https://github.com/ludlows/PESQ

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #1223
Add support for multiprocessing in `PESQ` metric for batched input.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/audio/pesq.py [hunk 1]
```diff
@@ -94,6 +96,9 @@ def __init__(
         if mode not in ("wb", "nb"):
             raise ValueError(f"Expected argument `mode` to either be 'wb' or 'nb' but got {mode}")
         self.mode = mode
+        if not isinstance(n_processes, int) and n_processes < 0:
```
Comment (Borda) [inline src/torchmetrics/audio/pesq.py:99]:
```suggestion
        if not isinstance(n_processes, int) and n_processes <= 0:
```

File: src/torchmetrics/audio/pesq.py [hunk 2]
```diff
@@ -42,7 +42,8 @@ class PerceptualEvaluationSpeechQuality(Metric):
         fs: sampling frequency, should be 16000 or 8000 (Hz)
         mode: ``'wb'`` (wide-band) or ``'nb'`` (narrow-band)
         keep_same_device: whether to move the pesq value to the device of preds
-
+        n_processes: integer specifiying the number of processes to run in parallel for the metric calculation.
+            Only applies to batches of data and if `multiprocessing` package is installed.
```
Comment (Borda) [inline src/torchmetrics/audio/pesq.py:46]:
```suggestion
            Only applies to batches of data and if ``multiprocessing`` package is installed.
```

**Review by stancld — APPROVED** [review]
Except for two @Borda's comments, LGTM. 👍

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki seems flaky time/speed-up tests for the latest Mac... :otter:

**Pull request merged**